corehq/apps/translations/app_translations/upload_form.py

```diff
@@ -158,13 +158,13 @@ def _add_or_remove_translations(self, lang, row):
         label_id = row['label']
         translations = self._get_translations_for_row(row, lang)
         translation_node = self.itext.find("./{f}translation[@lang='%s']" % lang)
-        keep_value_node = any(v for k, v in translations.items())
+        keep_value_node = any(v for k, v in translations.items()) or set(self.langs) != set(self.app.langs)
```

Comment:
i think we should put this in a variable with a nice name like
`all_app_langs_in_upload = set(self.langs) != set(self.app.langs)` 

Also better to keep this before the translations iterations?

Reply:
I was thinking about setting a flag at the class level to indicate whether this is single-sheet or multi-sheet - adding in 31aecb9c9314389055242c32512379ef5d196b9e. Doing that makes this clearer, I think.

Also swapped the order of the conditions.